browser/android/background_video/background_video_playback_tab_helper.cc

```diff
@@ -19,17 +19,56 @@
 #include "url/gurl.h"
 
 namespace {
-constexpr char16_t kYoutubeBackgroundPlaybackScript[] =
-    u"(function() {"
-    "    if (document._addEventListener === undefined) {"
-    "        document._addEventListener = document.addEventListener;"
-    "        document.addEventListener = function(a,b,c) {"
-    "            if(a != 'visibilitychange') {"
-    "                document._addEventListener(a,b,c);"
-    "            }"
-    "        };"
-    "    }"
-    "}());";
+constexpr char16_t kYoutubeBackgroundPlaybackAndPipScript[] =
+    uR"(
+(function() {
+    if (document._addEventListener === undefined) {
+        document._addEventListener = document.addEventListener;
+        document.addEventListener = function(a,b,c) {
+            if(a != 'visibilitychange') {
+                document._addEventListener(a,b,c);
+            }
+        };
+    }
+}());
+// Function to modify the flags if the target object exists.
+function modifyYtcfgFlags() {
+  if (!window.ytcfg) {
+    return;
+  }
+  const config = window.ytcfg.get("WEB_PLAYER_CONTEXT_CONFIGS")?.WEB_PLAYER_CONTEXT_CONFIG_ID_MWEB_WATCH
+  if (config && config.serializedExperimentFlags) {
+    let flags = config.serializedExperimentFlags;
+
+    // Replace target flags.
+    flags = flags
+      .replace("html5_picture_in_picture_blocking_ontimeupdate=true", "html5_picture_in_picture_blocking_ontimeupdate=false")
+      .replace("html5_picture_in_picture_blocking_onresize=true", "html5_picture_in_picture_blocking_onresize=false")
+      .replace("html5_picture_in_picture_blocking_document_fullscreen=true", "html5_picture_in_picture_blocking_document_fullscreen=false")
+      .replace("html5_picture_in_picture_blocking_standard_api=true", "html5_picture_in_picture_blocking_standard_api=false")
+      .replace("html5_picture_in_picture_logging_onresize=true", "html5_picture_in_picture_logging_onresize=false");
+
+    // Assign updated flags back to config.
+    config.serializedExperimentFlags = flags;
+    if (observer) {
+      observer.disconnect();
+    }
+  }
+}
+const observer = new MutationObserver((mutations) => {
+  for (const mutation of mutations) {
```

Comment:
MutationObservers can have a significant impact on performance, has this been reviewed by @brave/perf-team ?

Reply:
https://bravesoftware.slack.com/archives/C3T9S9WKD/p1744711045992409

Performance review team has been pinged on Slack!

Reply:
The problem with MutationObserver performance happens when we have a huge number of the observer callback hits. This can slow down the page. Usually we should make the filter the specific as possible.
Looks like here there no way to make filter specific, so we have to count and estimate the maximum number of `mutations` we will process.

WDYT about tracking `onload` event instead, does it work for you?
```
document.addEventListener('load', (event) => {
    const target = event.target;
    if (target.tagName === 'SCRIPT') {
      console.log('added', target.getAttribute('src'));
    }
  }, true);
  ```

P.S. the current impl with MutationObserver is likely to miss a case when added `node` isn't `<script>`, but has `<script>` in the subtree.

Reply:
Thank you @atuchin-m, the suggested solution works good! I'm going to commit the new changes.